### PR TITLE
llvm_bitreverse_i32

### PR DESCRIPTION
**Pull request by juj**
Implement llvm_bitreverse_i32() which was recently introduced to upstream LLVM at http://reviews.llvm.org/D14569.

### CODE REVIEW

**Comment by kripken** [comment]
Logic looks good. lgtm with a test in `test/cases`.

File: src/library.js [hunk 1]
```diff
@@ -1300,6 +1300,17 @@ LibraryManager.library = {
 
   llvm_dbg_declare__inline: function() { throw 'llvm_debug_declare' }, // avoid warning
 
+  llvm_bitreverse_i32__asm: true,
+  llvm_bitreverse_i32__sig: 'ii',
+  llvm_bitreverse_i32: function(x) {
+    x = x|0;
```
Comment (socketpair) [inline src/library.js]: add spaces around `|`

File: src/library.js [hunk 2]
```diff
@@ -1300,6 +1300,17 @@ LibraryManager.library = {
...
+    x = (((x & 0xaaaaaaaa) >> 1) | ((x & 0x55555555) << 1));
+    x = (((x & 0xcccccccc) >> 2) | ((x & 0x33333333) << 2));
+    x = (((x & 0xf0f0f0f0) >> 4) | ((x & 0x0f0f0f0f) << 4));
+    x = (((x & 0xff00ff00) >> 8) | ((x & 0x00ff00ff) << 8));
+    return((x >> 16) | (x << 16));
```
Comment (socketpair) [inline src/library.js]: add space after `return`

**Comment by kripken** [comment]
Thanks for the notes @socketpair.

File: src/library.js (hunk 2 above)
Comment (socketpair) [inline src/library.js]: Bug! Seems you should use >>> instead of >>. Does not it ?

Comment (socketpair) [inline src/library.js]: `llvm_bitreverse_i32(275600565)` gives `-8`.
`llvm_bitreverse_i32(-8)` gives `-1`

Surprise!

Comment (socketpair) [inline src/library.js]: @kripken 

```
llvm_bswap_i16: function(x) {
    x = x|0;
    return (((x&0xff)<<8) | ((x>>8)&0xff))|0;
  },
```
 seems same issue and so on

Comment (kripken) [inline src/library.js]: Good catch. Not all lines need it, but several do, so seems best to put it on them all, I agree.

@juj, seems good to add @socketpair's testcase here to the test when you get to it.

File: src/library.js (hunk 1 above)
Comment (juj) [inline src/library.js]: The existing convention is to omit spaces around `|0` operator that is used for asm.js function input type annotation, so keeping this as is.

**Comment by juj** [comment]
Thanks, revised with the correct shift operator and the test. Can you take a peek again @kripken that I got the asm.js input layout etc. proper in the `.ll` file?

File: src/library.js (hunk 2 above)
Comment (juj) [inline src/library.js]: Thanks for catching this @socketpair ! I had not had time to finalize this PR yet, updated it now with your numbers in the test case.

**Comment by kripken** [comment]
lgtm.

**Pull request merged**